build-support/bin/check_banned_imports.py

```diff
@@ -40,13 +42,18 @@ def find_files(*directories: str, extension: str) -> List[str]:
   ]
 
 
-def check_banned_import(files: List[str], *, bad_import_regex: str, correct_import_message: str) -> None:
-  regex = re.compile(bad_import_regex)
-  bad_files: List[str] = []
+def filter_files(files: List[str], *, snippet_regex: str) -> List[str]:
```

Comment:
Might be worth a docstring to note whether this returns a list with the files that were matched, or the ones that weren't.